### PR TITLE
Allow the definition of point in region plane using variables

### PR DESCRIPTION
**Pull request by evoyiatzis**
**Summary**

Allow the definition of point in region plane using variables 

**Related Issue(s)**

No related issue

**Author(s)**

E. Voyiatzis at NovaMechanics Ltd (email evoyiatzis@gmail.com)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

It does not break backward compatibility 

**Implementation Notes**

The modifications in the code at region_plane.h and region_plane.cpp follow the approach in region_sphere.cpp and region_sphere.h A xstyle/ystyle/zstyle int variable stores if the point is defined as constant or via variable and the xstr/ystr/zstr stores the name of the variable if required. The init method calls variable_check to check if the definition and the numerical value of the point is sound. The deconstructor cleans the memory from the xstr/ystr/zstr variables. 

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [ ] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] One or more example input decks are included

**Further Information, Files, and Links**
A set of test files is included in the attached zip file

[equilibrated_germanene_sheet_50.0.zip](https://github.com/user-attachments/files/16780858/equilibrated_germanene_sheet_50.0.zip)

### CODE REVIEW

File: doc/src/region.rst [hunk 1]
```diff
@@ -34,10 +34,11 @@ Syntax
        *ellipsoid* args = x y z a b c
          x,y,z = center of ellipsoid (distance units)
          a,b,c = half the length of the principal axes of the ellipsoid (distance units)
-           x,y,z,a,b and c can be a variable (see below)
+         x,y,z,a,b and c can be a variable (see below)
        *plane* args = px py pz nx ny nz
          px,py,pz = point on the plane (distance units)
          nx,ny,nz = direction normal to plane (distance units)
+         px,py,pz can be a variable
```
Comment (akohlmey) [inline doc/src/region.rst]:
```suggestion
           px,py,pz can be a variable
```
The list is more readable when the "can be a variable" statements are indented.

File: doc/src/region.rst [hunk 2]
```diff
@@ -203,12 +204,13 @@ and with radius as its radius.
 
 The *radius* value for styles *sphere* and *cylinder*, and the
 parameters a,b,c for style *ellipsoid*, can each be specified as an
-equal-style :doc:`variable <variable>`.  Likewise, for style *sphere*
```
Comment (akohlmey) [inline doc/src/region.rst:206]: For better readability for the documentation source, there should be two blanks after a full stop. We usually add them when revising documentation. There is a lot of legacy text without. Removing those extra blanks just creates extra work for the folks maintaining the manual.

File: doc/src/region.rst (hunk 1 above)
Comment (evoyiatzis) [reply doc/src/region.rst]: @akohlmey  thanks for the explanation !

File: doc/src/region.rst [hunk 3]
```diff
@@ -18,13 +18,13 @@ Syntax
        *delete* = no args
        *block* args = xlo xhi ylo yhi zlo zhi
          xlo,xhi,ylo,yhi,zlo,zhi = bounds of block in all dimensions (distance units)
-         xlo,xhi,ylo,yhi,zlo,zhi can be a variable
+           xlo,xhi,ylo,yhi,zlo,zhi can be a variable
```
Comment (akohlmey) [inline doc/src/region.rst]:
```suggestion
           xlo,xhi,ylo,yhi,zlo,zhi can be a variable (see below)
```

File: doc/src/region.rst [hunk 4]
```diff
@@ -38,6 +38,7 @@ Syntax
        *plane* args = px py pz nx ny nz
          px,py,pz = point on the plane (distance units)
          nx,ny,nz = direction normal to plane (distance units)
+           px,py,pz can be a variable
```
Comment (akohlmey) [inline doc/src/region.rst]:
```suggestion
           px,py,pz can be a variable (see below)
```

**Review by jmgoff — APPROVED** [review]

**Pull request merged**